Switch default memory format of _like operators to Preserve

VitalyFedyunin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #31227 Fix `.to` operator to preserve strides when `self` should be returned
* #31226 Revert "Switch default memory format of to (and similar) operators to Preserve"
* #31139 [WIP] Fix cudnn channels_last descriptio problem
* #31131 Add memory_format support to `zeros`, `ones`, `full` (still need tests)
* #30089 Switch default memory format of clone operator to Preserve
* #30088 Switch default memory format of to (and similar) operators to Preserve
* **#30087 Switch default memory format of _like operators to Preserve**

Differential Revision: [D18624986](https://our.internmc.facebook.com/intern/diff/D18624986)

VitalyFedyunin committed:
Switch default memory format of _like operators to Preserve

[ghstack-poisoned]

VitalyFedyunin committed:
rebase to master + fix tests on "Switch default memory format of _like operators to Preserve"




[ghstack-poisoned]

VitalyFedyunin committed:
fix tests code on "Switch default memory format of _like operators to Preserve"


Differential Revision: [D18624986](https://our.internmc.facebook.com/intern/diff/D18624986)

[ghstack-poisoned]

VitalyFedyunin commented:
@ailzhang can you please take a look at XLA failure

ailzhang commented:
@VitalyFedyunin XLA is passing, no? ;)

VitalyFedyunin committed:
add experimental code to check model on "Switch default memory format of _like operators to Preserve"


Differential Revision: [D18624986](https://our.internmc.facebook.com/intern/diff/D18624986)

[ghstack-poisoned]

VitalyFedyunin committed:
experiments with tag on "Switch default memory format of _like operators to Preserve"


Differential Revision: [D18624986](https://our.internmc.facebook.com/intern/diff/D18624986)

[ghstack-poisoned]

VitalyFedyunin committed:
rebase to master on "Switch default memory format of _like operators to Preserve"


Differential Revision: [D18624986](https://our.internmc.facebook.com/intern/diff/D18624986)

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit f669291:
* **1/1** failures introduced in this PR
* **1/1** recognized as flaky
    * Re-run these jobs?



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=f669291cc982c520cf02d749d88945e2ce28a962).

### 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3923969) caffe2_onnx_py3_6_clang7_ubuntu16_04_build (1/1)
**Step:** "Set Up CI Environment After attach_workspace" ([details](https://dr.pytorch.org/build-details.html?build_id=73298472))
```

93% [26 moreutils 12.3 kB/55.0 kB 22%]
                                      
94% [Working]
             
Get:27 http://us-central1.gce.archive.ubuntu.com/ubuntu xenial/main amd64 tcl8.6 amd64 8.6.5+dfsg-2 [14.2 kB]

94% [27 tcl8.6 14.2 kB/14.2 kB 100%]
                                    
94% [Working]
             
Get:28 http://us-central1.gce.archive.ubuntu.com/ubuntu xenial-updates/main amd64 thermald amd64 1.5-2ubuntu4 [187 kB]

94% [28 thermald 8,042 B/187 kB 4%]
                                   
95% [Working]
             
Get:29 http://us-central1.gce.archive.ubuntu.com/ubuntu xenial/main amd64 tk8.6 amd64 8.6.5-1 [12.2 kB]

95% [29 tk8.6 12.2 kB/12.2 kB 100%]
                                   
96% [Working]
             
Get:30 http://us-central1.gce.archive.ubuntu.com/ubuntu xenial/main amd64 xterm amd64 322-1ubuntu1 [607 kB]

96% [30 xterm 12.3 kB/607 kB 2%]
                                
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]                                                      20.9 MB/s 0s
97% [Working]                                                      20.9 MB/s 0s
97% [Working]                                                      20.9 MB/s 0s
97% [Working]                                                      20.9 MB/s 0s
97% [Working]                                                      20.9 MB/s 0s
97% [Working]                                                      20.9 MB/s 0s
97% [Working]                                                      20.9 MB/s 0s
97% [Working]                                                      20.9 MB/s 0s
97% [Working]                                                      20.9 MB/s 0s
97% [Working]                                                      20.9 MB/s 0s
97% [Working]                                                      20.9 MB/s 0s
97% [Working]                                                      20.9 MB/s 0s
                                                                               
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
97% [Working]
             
Err:31 https://nvidia.github.io/nvidia-container-runtime/ubuntu16.04/amd64  nvidia-container-runtime-hook 1.4.0-1
  Operation too slow. Less than 10 bytes/sec transferred the last 120 seconds

97% [Working]
             
Get:32 https://nvidia.github.io/nvidia-container-runtime/ubuntu16.04/amd64  nvidia-container-runtime 2.0.0+docker18.09.4-1 [1,758 kB]

97% [32 nvidia-container-runtime 4,867 B/1,758 kB 0%]         802 B/s 48min 26s
98% [Working]                                                  802 B/s 12min 0s
                                                                               
Get:33 https://nvidia.github.io/nvidia-docker/ubuntu16.04/amd64  nvidia-docker2 2.0.3+docker18.09.4-1 [2,882 B]

98% [33 nvidia-docker2 2,882 B/2,882 B 100%]                  802 B/s 11min 56s
99% [Working]                                                 802 B/s 11min 56s
                                                                               
Fetched 130 MB in 2min 0s (1,081 kB/s)
W: --force-yes is deprecated, use one of the options starting with --allow instead.
E: Failed to fetch https://nvidia.github.io/nvidia-container-runtime/ubuntu16.04/amd64/./nvidia-container-runtime-hook_1.4.0-1_amd64.deb  Operation too slow. Less than 10 bytes/sec transferred the last 120 seconds

E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
```


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

gchanan commented on aten/src/ATen/native/TensorFactories.cpp:
```diff
@@ -251,7 +251,7 @@ Tensor empty_like(
 
   auto memory_format =
-      optional_memory_format.value_or(MemoryFormat::Contiguous);
+      optional_memory_format.value_or(MemoryFormat::Preserve);
```
are you going to change new_empty (below)?  I don't know what the right answer is, just asking.

VitalyFedyunin replied:
Eventually yes, as we start increasing support.

VitalyFedyunin committed:
Fix flake8 on "Switch default memory format of _like operators to Preserve"


Differential Revision: [D18624986](https://our.internmc.facebook.com/intern/diff/D18624986)

[ghstack-poisoned]

VitalyFedyunin committed:
Fix flake8 on "Switch default memory format of _like operators to Preserve"


Differential Revision: [D18624986](https://our.internmc.facebook.com/intern/diff/D18624986)

[ghstack-poisoned]
